src/core/nc_device.c

```diff
@@ -318,16 +318,17 @@ static np_error_code nc_device_populate_mdns(struct nc_device_context* device)
     }
 
     char uniqueId[64];
+    memset(uniqueId, 0, 64);
     if (strlen(device->productId) + 1 + strlen(device->deviceId) > 63) {
         return NABTO_EC_INVALID_STATE;
     }
 
-    char* ptr = uniqueId;
-    strcpy(ptr, device->productId);
-    ptr += strlen(device->productId);
-    strcpy(ptr, "-");
-    ptr += strlen("-");
-    strcpy(ptr, device->deviceId);
+    if (!nn_strcat(uniqueId, 64, device->productId) ||
+        !nn_strcat(uniqueId, 64, "-") ||
+        !nn_strcat(uniqueId, 64, device->deviceId)) {
+        return NABTO_EC_OUT_OF_MEMORY;
```

Comment:
This is not a memory allocation problem, I think NABTO_EC_OUT_OF_MEMORY generally means an allocation error occured.